srv/salt/ceph/monitoring/alertmanager/default.sls

```diff
@@ -4,8 +4,45 @@ install alertmanager:
     - fire_event: True
     - refresh: True
 
+{% set receiver_snmp_enabled = salt['pillar.get']('monitoring:alertmanager:receiver:snmp:enabled', False) %}
+{% if receiver_snmp_enabled | to_bool %}
+
+prometheus_webhook_snmp_install:
+  pkg.installed:
+    - name: prometheus-webhook-snmp
+    - fire_event: True
+    - refresh: True
+
+prometheus_webhook_snmp_configure:
+  file.managed:
+    - name: "/etc/prometheus-webhook-snmp.conf"
+    - source: salt://ceph/monitoring/alertmanager/files/prometheus-webhook-snmp.conf.j2
+    - template: jinja
+    - context:
+        config: {{ salt['pillar.get']('monitoring:alertmanager:receiver:snmp:config', {}) | json }}
```

Comment:
Alternatively to having a template file, you could also use the pillar straight as the content:
```
content: |
    {{salt['pillar.get']('monitoring:alertmanager:receiver:snmp:config', {}) | json}}
```

Reply:
also any reason you use json here when the config file is in yaml?

Reply:
That was my first approach:
```
contents: {{ salt['pillar.get']('monitoring:alertmanager_receiver_snmp:config', {}) | yaml }}
```
which did not produce the expected output:
```
OrderedDict([('foo', 'bar'), ('metrics', True)])
```
Using 
```
prometheus_webhook_snmp_configure:
  file.managed:
    - name: "/etc/prometheus-webhook-snmp.conf"
    - contents: |
        {{ salt['pillar.get']('monitoring:alertmanager_receiver_snmp:config', {}) | yaml }}
```
also does not output what i want to have:
```
{foo: bar, metrics: true}
```
This looks much better, but i want to have
```
foo: bar
metrics: true
```
That's why i use the Jinja template to iterate over the key/values.

Reply:
sigh...you right. I guess it would have been surprising if something was easy in salt.
The `yaml` filter does work but only with the correct indentation. Like so:
```
- contents: |
    {{ salt['pillar.get']('monitoring:alertmanager:receiver:snmp:config', {}) | yaml(False) | indent(8) }}
```
The value indent gets depends on the indent level in the file. I think 8 should be right in `srv/salt/ceph/monitoring/alertmanager/default.sls`

Reply:
eh I think that might work but only for this case...feel free to ignore while I quietly hate salt.

Reply:
Fixed. I've chosen this solution because it requires one file less.